submissions/eoneoff/port-sniffer/sniffer.js

```diff
@@ -0,0 +1,198 @@
+            or set incorrectly the default range of (1-65535) is scanned
+
+                node sniffer.js --host wwww.google.com --ports 1024-2018
+
+--help      parameter to see this reference
+
+To import port scanner use
+
+    const sniffer = require('sniffer.js')
+
+after import you can perform scan by calling sniffer.scan()
+
+sniffer.scan(host[, portString][, args])
+
+For detailed disambiguation on the function call sniffer.help()`,
+function: `Simple port scanning method
+
+This is an async function for port scanning. It takes one obligatory argument 'host' and two
+optional arguments 'portString', and additional arguments object 'args' with 'module' and
+'silent' properties.
+
+    sniffer.scan(host[, portString][, args])
+
+host:str        an obligatory argument, which provies a host for scanning. It can be an
+                IP address as string of four numbers, separated by dots or a hostname
+
+posrtString:str an optional argument to define a range of scanned ports. Should be provided
+                as a string, composed of two numbers, separated by dash. If not provided or
+                provided incorrectly the port range will be set to default full range (1-65535)
+
+args:Object                
+
+    module:bool     a bool argument, which tells the function to run as a module, that is to
+                    return a result as a list of open ports
+
+    silent:bool     a bool argument, which tells the function to supress the output and run
+                    in silent mode. Takes effect only if 'modlue' parameter is also set
+                    to 'true'
+
+The function is async and returns a promice. You shoud eather use it with await operator or
+with then() method`
+}
+
+const dns = require('dns').promises;
+const Socket = require('net').Socket;
+
+function* twister(){
+    let symbols = ['|','/','-','\\'];
+    let current = 0;
+    while (true) {
+        if (current == 4) current = 0;
+        yield symbols[current++];
+    }
+}
+
+function isIpAddress(address){
+    return address.search(/^((25[0-5]|2[0-4]\d|[01]?\d?\d)\.){3}(25[0-5]|2[0-4]\d|[01]?\d?\d)$/) == -1;
+}
+
+function isRange(ip) {
+    return ip.search(/\d{1,4}-\d{1,4}/) == -1;
+}
+
+function isInvalidPortsRange(portsRange){
+    return +portsRange[0] >= +portsRange[1] || +portsRange[0] < 1 || +portsRange[0] > 65535
+    || +portsRange[1] < 1 || +portsRange[1]>65535;
+}
+
+async function parseHost(hostName){
+    let host = '';
+    if(isIpAddress(hostName)){
+        try{
+            host = (await dns.lookup(hostName)).address;
+        } catch {
+            throw "Hostname can not be resolved";
+        }
+    } else {
+        host = hostName;
+    }
+    return host;
+}
+
+function parsePorts(ports){
+    if(isRange(ports)){
+        throw "Invalid port range argument"
+    } else {
+        let limits = ports.split("-");
+        if (isInvalidPortsRange(limits)){
+                throw "Port numbers out of range"
+            } else {

```

Comment:
same as above `throw` will force function to stop execution

Reply:
It seems to me, that just continuing the code in one block, it will make a little obscure, that those are two separate brunches of code.

Reply:
is nothing special here, `throw` is similar to `return`